File: algorithm/greedy/change/change.py
# 거스름 돈
n = int(input("거스름돈: "))
# 동전 개수
cnt = 0
# 동전 종류
coin = [500, 100, 50, 10]

for c in coin:
  if n >= c:
    cnt += n // c
    n %= c

print(cnt)

File: algorithm/greedy/change/Change.java
import java.util.Scanner;

public class Change {
  public static void main(String[] args) {

    Scanner sc = new Scanner(System.in);
    System.out.println("java 거스름돈: ");
    int n = sc.nextInt();
    int cnt = 0;
    int[] coin = {500, 100, 50, 10};
    
    for (int i = 0; i < 4; i++){
      if(n >= coin[i]) {
        cnt += n / coin[i];
        n %= coin[i];
      }
    }

    System.out.println(cnt);
  }
}